File: pages/cart_page.py
from playwright.sync_api import Page, expect

from pages.base_page import BasePage


class CartPage(BasePage):
    def __init__(self, page: Page):
        super().__init__(page)
        self.cart_items = page.locator(".cart_item")
        self.checkout_button = page.locator('[data-test="checkout"]')

    def expect_product_visible(self, product_name: str) -> None:
        expect(self.cart_items.filter(has_text=product_name)).to_be_visible()

    def expect_item_count(self, count: int) -> None:
        expect(self.cart_items).to_have_count(count)

    def proceed_to_checkout(self) -> None:
        self.checkout_button.click()

File: pages/cart.page.ts
import { expect } from '@playwright/test';
import { BasePage } from './base.page';

export class CartPage extends BasePage {
  private readonly cartItems = this.page.locator('.cart_item');
  private readonly checkoutButton = this.page.locator('[data-test="checkout"]');

  async expectProductVisible(productName: string) {
    await expect(this.cartItems.filter({ hasText: productName })).toBeVisible();
  }

  async expectItemCount(count: number) {
    await expect(this.cartItems).toHaveCount(count);
  }

  async proceedToCheckout() {
    await this.checkoutButton.click();
  }
}
